Bugfix/zenko 2034 crr issue

rahulreddy opened the pull request:
- test: ZENKO-2034 expand test to assert location information
- bugfix: ZENKO-2034 set location info for oob objects
The location info needs to be set as if the object has been written
to the bucket which has S3C as it's backend. This allows GETs to happen
on the object with data calls to S3C over S3 api.

rahulreddy committed:
test: ZENKO-2034 expand test to assert location information

bert-e commented:
# Hello rahulreddy,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/backbeat/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @bennettbuchanan

* @vrancurel

* @rahulreddy

* @jonathan-gramain

JianqinWang reviewed (approved):
fix was verified on Zenko 1.1 deployment with 7.4.4 S3C

jonathan-gramain commented on extensions/mongoProcessor/MongoQueueProcessor.js:
```diff
@@ -280,19 +281,18 @@ class MongoQueueProcessor {
+                start: 0,
+                dataStoreName: zenkoLocation,
+                dataStoreType: 'aws_s3',
+                dataStoreETag: entry.getContentMd5(),
```
Shouldn't it be `1:${entry.getContentMd5()}`?

rahulreddy replied:
True.

rahulreddy replied:
Just checked `ObjectMDLocation` model. You are right.

jonathan-gramain replied:
what about actually using the model :wink: not a blocker though

rahulreddy replied:
A refactor is coming 🔜

jonathan-gramain commented on tests/functional/ingestion/MongoQueueProcessor.js:
```diff
@@ -369,11 +369,36 @@ describe('MongoQueueProcessor', function mqp() {
+                start: 0,
+                size: 50,
+                dataStoreName: LOCATION,
+                dataStoreETag: '1-7cccfcef3abdfaba48b2d193cb146074',
```
I don't remember why we have `1-` here and not `1:` but maybe it's normal.

rahulreddy replied:
Mistake but it is ignored in the code.

rahulreddy committed:
bugfix: ZENKO-2034 set location info for oob objects

The location info needs to be set as if the object has been written
to the bucket which has S3C as it's backend. This allows GETs to happen
on the object with data calls to S3C over S3 api.

rahulreddy commented:
@bert-e approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/8.1`



The following branches will **NOT** be impacted:

* `development/7.4`
* `development/7.5`
* `development/8.0`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/8.1`



The following branches have **NOT** changed:

* `development/7.4`
* `development/7.5`
* `development/8.0`



Please check the status of the associated issue ZENKO-2034.

Goodbye rahulreddy.
